src/App.tsx

```diff
@@ -15,48 +19,123 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort(
+        (prevGood, currGood) => prevGood.localeCompare(currGood),
+      );
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods.sort(
+        (prevGood, currGood) => prevGood.length - currGood.length,
+      );
+      break;
+
+    default:
+      break;
+  }
+
+  return isReversed
+    ? visibleGoods.reverse()
+    : visibleGoods;
+}
```

Comment:
Here you could write shorter:
  if (isReversed) {
    visibleGoods.reverse();
  }